embedded-tests/src/test/java/org/apache/druid/testing/embedded/docker/DruidContainerResource.java

```diff
@@ -0,0 +1,310 @@
+        .withCommonProperties(getCommonProperties())
+        .withServiceProperties(getServerProperties())
+        .withFileSystemBind(
+            segmentDeepStorageDirectory.hostPath,
+            segmentDeepStorageDirectory.containerPath,
+            BindMode.READ_WRITE
+        )
+        .withFileSystemBind(
+            indexerLogsDeepStorageDirectory.hostPath,
+            indexerLogsDeepStorageDirectory.containerPath,
+            BindMode.READ_WRITE
+        )
+        .withFileSystemBind(
+            serviceLogsDirectory.hostPath,
+            serviceLogsDirectory.containerPath,
+            BindMode.READ_WRITE
+        )
+        .withEnv(
+            Map.of(
+                "DRUID_SET_HOST_IP", "0",
+                "DRUID_SET_HOST", "0"
+            )
+        )
+        .waitingFor(Wait.forHttp("/status/health").forPort(servicePort));
+
+    // Bind the ports statically (rather than using a mapped port) to the same
+    // value used in `druid.plaintextPort` to make this node discoverable
+    // by other services (both embedded and dockerized).
+    List<String> portBindings = exposedPorts.stream().map(
+        port -> StringUtils.format("%d:%d", port, port)
+    ).collect(Collectors.toList());
+    container.setPortBindings(portBindings);
+
+    return container;
+  }
+
+  /**
+   * Writes the common properties to a file.
+   *
+   * @return Name of the file.
+   */
+  private Properties getCommonProperties()
+  {
+    final Properties commonProperties = new Properties();
+    commonProperties.putAll(cluster.getCommonProperties());
+    FORBIDDEN_PROPERTIES.forEach(commonProperties::remove);
+
+    commonProperties.setProperty(
+        "druid.storage.storageDirectory",
+        segmentDeepStorageDirectory.containerPath
+    );
+    commonProperties.setProperty(
+        "druid.indexer.logs.directory",
+        indexerLogsDeepStorageDirectory.containerPath
+    );
+
+    log.info(
+        "Writing common properties for Druid container[%s]: [%s]",
+        name, commonProperties
+    );
+
+    return commonProperties;
+  }
+
+  /**
+   * Writes the server properties to a file.
+   *
+   * @return Name of the file.
+   */
+  private Properties getServerProperties()
+  {
+    FORBIDDEN_PROPERTIES.forEach(properties::remove);
+    addProperty("druid.host", EmbeddedDruidCluster.getDefaultHost());
+    addProperty("druid.plaintextPort", String.valueOf(servicePort));
+
+    final Properties serverProperties = new Properties();
+    serverProperties.putAll(properties);
+
+    log.info(
+        "Writing runtime properties for Druid container[%s]: [%s]",
+        name, serverProperties
+    );
+
+    return serverProperties;
+  }
+
+  private static void createLogDirectory(File dir)
+  {
+    try {
+      FileUtils.mkdirp(dir);
+      Files.setPosixFilePermissions(dir.toPath(), PosixFilePermissions.fromString("rwxrwxrwx"));
+    }
+    catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
```

Comment:
couldn't the `MountedDir` provide a place for this method? that way if someone needs similar in the future it might got reused...

Reply:
Updated ,thanks for the suggestion!